server/api/subscriptions.go

```diff
@@ -84,14 +84,19 @@ func (a *API) handleCreateSubscription(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// check for valid block
-	_, bErr := a.app.GetBlockByID(sub.BlockID)
+	// check for valid block and permissions to view its board
+	block, bErr := a.app.GetBlockByID(sub.BlockID)
 	if bErr != nil {
 		message := fmt.Sprintf("invalid blockID: %s", bErr)
 		a.errorResponse(w, r, model.NewErrBadRequest(message))
 		return
 	}
 
+	if !a.permissions.HasPermissionToBoard(sub.SubscriberID, block.BoardID, model.PermissionViewBoard) {
```

Comment:
Is the same check also needed in `handleDeleteSubscription` and `handleGetSubscriptions` APIs?

Reply:
No, delete/get are self-only. User can not be able to delete or get other users subscription so not boards check required here.